### PR TITLE
Command bar refactor

### PR DESCRIPTION
**Pull request by micahgodbolt**
#### Pull request checklist

Preview of CommandBar updates

### CODE REVIEW

File: packages/example-app-base/src/components/ExampleCard/ExampleCard.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import * as React from 'react';
 import { css } from 'office-ui-fabric-react/lib/Utilities';
 import './ExampleCard.scss';
-import { CommandButton } from 'office-ui-fabric-react/lib/Button';
+import { ActionButton } from 'office-ui-fabric-react/lib/Button';
```
Comment (dzearing) [inline packages/example-app-base/src/components/ExampleCard/ExampleCard.tsx]: hmmm... you might be doing too much in one PR :)

File: packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx [hunk 2]
```diff
@@ -1,413 +1,162 @@
...
+                this._onRenderSearch(this.props)
+              }
+
+              {/*Primary Items*/ }
+              <OverflowSet
+                className={ css(styles.primarySet) }
+                items={ data.primaryItems }
+                overflowItems={ data.overflowItems.length ? data.overflowItems : null }
+                onRenderItem={ onRenderItems }
+                onRenderOverflowButton={ (renderedOverflowItems: ICommandBarItemProps[]) => {
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx]: shouldn't the default button be good enough here? Do you really need to implement an override?

File: packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx [hunk 3]
```diff
@@ -1,413 +1,162 @@
...
-
-      if (!this._commandItemWidths[item.key]) {
-        let el = this.refs[item.key] as HTMLElement;
-
-        if (el) {
-          this._commandItemWidths[item.key] = el.getBoundingClientRect().width;
-        }
-      }
-    }
+    // this.refs.focusZone.focus();
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx]: hmm

File: packages/example-app-base/src/components/ExampleCard/ExampleCard.tsx (hunk 1 above)
Comment (micahgodbolt) [inline packages/example-app-base/src/components/ExampleCard/ExampleCard.tsx]: I agree, I could break this action button into a different PR, but it's a big breaking change, and it's needed for the command bar, so I'd need to merge action button into a 4.0 branch and then merge it right back into command bar anyway. :)

File: packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx (hunk 2 above)
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx]: valid question. i'd had them split out in overflow set, as they certainly take a unique set of props.

File: packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx (hunk 3 above)
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx]: yup :) need to dig into what this did in the original so that I can replicate it.
